### PR TITLE
[ETL-287] JSON to Parquet job

### PR DESCRIPTION
**Pull request by philerooski**
* Implementation of JSON to Parquet job
* There is one JSON to Parquet job per data type. I spent some time trying to get the trigger to start many instances of the same job with different (`--glue-table`) arguments, but as it turns out triggers cannot start more than one instance of the same job.
* JSON and Parquet datasets are now written to a namespaced S3 prefix.
* The S3 to JSON and JSON to Parquet jobs are not namespaced, although the workflow they run within is.
* The Glue workflow was modified so that the successful completion of the S3 to JSON job triggers every JSON to Parquet job.

### CODE REVIEW

**Review by rxu17 — COMMENTED** [review]
Otherwise, lgtm!

**Review by rxu17 — APPROVED** [review]

File: config/develop/namespaced/glue-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+template:
+  path: glue-workflow.j2
+stack_name: "{{ stack_group_config.namespace }}-glue-workflow"
+dependencies:
+  - develop/namespaced/glue-tables.yaml
+parameters:
+  Namespace: {{ stack_group_config.namespace }}
+  JsonBucketName: !stack_output_external recover-dev-intermediate-bucket::BucketName
+  ParquetBucketName: !stack_output_external recover-dev-processed-data-bucket::BucketName
```
[the remark this answers is not in the document]
Comment (philerooski) [reply config/develop/namespaced/glue-workflow.yaml:9]: Yes, that would be cleaner. Although this issue affects all of our stacks, not just the ones in this PR (for example, the `glue-job-S3ToJsonS3` stack calls the "IntermediateBucket" the "TempBucket". So I think it's a separate JIRA ticket worth of work.

File: templates/glue-job-JSONToParquet.j2 [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+Resources:
+
+  {% set datasets = [] %}
+  {% for v in sceptre_user_data.dataset_schemas.tables.keys() if not "Deleted" in v %}
+    {% set dataset = {} %}
+    {% do dataset.update({"type": v}) %}
+    {% do dataset.update({"table_name": "dataset_" + v.lower()})%}
+    {% do dataset.update({"stackname_prefix": "{}".format(v.replace("_",""))}) %}
+    {% do datasets.append(dataset) %}
+  {% endfor %}
```
[the remark this answers is not in the document]
Comment (philerooski) [reply templates/glue-job-JSONToParquet.j2:54]: I don't think there's a way to set Jinja "globals" that can be accessed from any stack.

File: config/develop/namespaced/glue-workflow.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (rxu17) [reply config/develop/namespaced/glue-workflow.yaml:9]: Yeah. :(

[the remark this answers is not in the document]
Comment (thomasyu888) [reply config/develop/namespaced/glue-workflow.yaml:9]: Lets make sure we create a ticket for that

**Review by thomasyu888 — APPROVED** [review]
🔥 Great work!

File: src/glue/jobs/json_to_parquet.py [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+                .dropDuplicates(
+                    subset=INDEX_FIELD_MAP[table_data_type]
+                )
+            ),
+            glue_ctx=glue_context,
+            name=table_name
+    )
+    return table
+
+def write_table_to_s3(
```
Comment (thomasyu888) [inline src/glue/jobs/json_to_parquet.py:131]: Are some of these functions used across BridgeDownstream and RECOVER?  I wonder if it'd be worth creating a package.  Can you install custom packages in AWS Glue?

File: src/glue/jobs/json_to_parquet.py [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+            return True
+    return False
+
+def get_table(
+        table_name: str,
+        database_name: str,
+        glue_context: GlueContext
+    ) -> DynamicFrame:
+    """
+    Return a table as a DynamicFrame with an unambiguous schema.
```
Comment (thomasyu888) [inline src/glue/jobs/json_to_parquet.py:92]: What do you mean by this? How does dropping the duplicates make the schema umambiguous?

File: config/develop/namespaced/glue-workflow.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (philerooski) [reply config/develop/namespaced/glue-workflow.yaml:9]: https://sagebionetworks.jira.com/browse/ETL-348

File: src/glue/jobs/json_to_parquet.py (hunk 3 above)
Comment (philerooski) [reply src/glue/jobs/json_to_parquet.py:131]: > Are some of these functions used across BridgeDownstream and RECOVER? I wonder if it'd be worth creating a package.
Yes, we will want to maintain a separate python library once we have a more generalized framework for data to parquet. I'm not sure if it makes sense doing this for RECOVER considering the tight deadline we are on, but it's worth keeping in mind for future projects that require a data to compute of some sort.

File: src/glue/jobs/json_to_parquet.py (hunk 4 above)
Comment (philerooski) [reply src/glue/jobs/json_to_parquet.py:92]: It doesn't. Dynamic Frames have an "ambiguous" schema in the sense that they do not enforce a schema on the individual Dynamic Records. To fix the schema (more specifically, to fix the data types), you need to call `resolveChoice` on the Dynamic Frame.

**Pull request merged**